src/assert_output.bash

```diff
@@ -137,6 +141,7 @@ assert_output() {
     case "$1" in
     -p|--partial) is_mode_partial=1; shift ;;
     -e|--regexp) is_mode_regexp=1; shift ;;
+    -d|--diff) show_diff=1; shift ;;
```

Comment:
Again, `--diff` might be a valid output to check for. At least, we have `--` here to differentiate options from output but we should at least check that there still is a trailing output.